services/rows/tests/routes/test_rows.py

```diff
@@ -377,34 +390,57 @@ def test_create_response(ds: Dataset, app_config: AppConfig, cached_assets_direc
 def test_create_response_with_image(
     ds_image: Dataset, app_config: AppConfig, cached_assets_directory: StrPath
 ) -> None:
-    response = create_response(
-        dataset="ds_image",
-        config="default",
-        split="train",
-        cached_assets_base_url=app_config.cached_assets.base_url,
-        cached_assets_directory=cached_assets_directory,
-        pa_table=ds_image.data,
-        offset=0,
-        features=ds_image.features,
-        unsupported_columns=[],
-        num_rows_total=10,
-    )
-    assert response["features"] == [{"feature_idx": 0, "name": "image", "type": {"_type": "Image"}}]
-    assert response["rows"] == [
-        {
-            "row_idx": 0,
-            "row": {
-                "image": {
-                    "src": "http://localhost/cached-assets/ds_image/--/default/train/0/image/image.jpg",
-                    "height": 480,
-                    "width": 640,
-                }
-            },
-            "truncated_cells": [],
-        }
-    ]
-    cached_image_path = Path(cached_assets_directory) / "ds_image/--/default/train/0/image/image.jpg"
-    assert cached_image_path.is_file()
+    dataset, config, split = "ds_image", "default", "train"
+    with mock_s3():
+        bucket_name = "bucket"
+        access_key_id = "access_key_id"
+        secret_access_key = "secret_access_key"
+        region = "us-east-1"
+        s3_client = S3Client(
+            region_name=region, aws_access_key_id=access_key_id, aws_secret_access_key=secret_access_key
+        )
+        folder_name = "cached-assets"
+        conn = boto3.resource("s3", region_name=region)
+        conn.create_bucket(Bucket=bucket_name)
+
+        with patch("rows.routes.rows.CACHED_ASSETS_S3_SUPPORTED_DATASETS", [dataset]):
+            response = create_response(
+                dataset=dataset,
+                config=config,
+                split=split,
+                cached_assets_base_url=app_config.cached_assets.base_url,
+                s3_client=s3_client,
+                cached_assets_directory=cached_assets_directory,
+                cached_assets_s3_bucket=bucket_name,
+                cached_assets_s3_folder_name=folder_name,
+                pa_table=ds_image.data,
+                offset=0,
+                features=ds_image.features,
+                unsupported_columns=[],
+                num_rows_total=10,
+            )
+        assert response["features"] == [{"feature_idx": 0, "name": "image", "type": {"_type": "Image"}}]
+        assert response["rows"] == [
+            {
+                "row_idx": 0,
+                "row": {
+                    "image": {
+                        "src": "http://localhost/cached-assets/ds_image/--/default/train/0/image/image.jpg",
```

Comment:
do we want to proxy the images through the Datasets Server? else: it should be the URL of the S3 bucket, or the CDN in front of it

Reply:
also: to make it easier to manage the response, possibly we need two keys:
1. `src`: the online resource
2. `path`: the path inside the storage (be it S3 or another filesystem)

Reply:
As discussed with @rtrompier once the CDN is enabled in prod, it will be transparent for us to use the current URL for cached-assets like for example: https://datasets-server.huggingface.co/cached-assets/asoria/image/--/default/train/0/image/image.jpg

> also: to make it easier to manage the response, possibly we need two keys

Not sure if it is a good time to add more fields to the response. How the new `path` field could be useful from the client side?

Reply:
OK. Indeed the path is something internal.